NLog.Windows.Forms/RichTextBoxTarget.cs

```diff
@@ -202,62 +234,127 @@ public RichTextBoxTarget()
+
+                if (FormName == null)
                 {
-                    bool error = false;
-                    TargetForm = openFormByName;
-                    if (string.IsNullOrEmpty(ControlName))
-                    {
-                        error = true;
-                        var message = "Rich text box control name must be specified for " + GetType().Name + ".";
-                        if (LogManager.ThrowExceptions)
-                        {
-                            throw new NLogConfigurationException(message);
-                        }
-
-                        InternalLogger.Error(message);
-                    }
+                    HandleError("FormName should be specified for " + GetType().Name + "." + this.Name);
+                    return;
+                }
 
-                    CreatedForm = false;
-                    TargetRichTextBox = FormHelper.FindControl<RichTextBox>(ControlName, TargetForm);
+                if (string.IsNullOrEmpty(ControlName))
+                {
+                    HandleError("Rich text box control name must be specified for " + GetType().Name + "." + this.Name);
+                    return;
+                }
 
-                    if (TargetRichTextBox == null)
-                    {
-                        error = true;
-                        var message = "Rich text box control '" + ControlName + "' cannot be found on form '" + FormName + "'.";
-                        if (LogManager.ThrowExceptions)
-                        {
-                            throw new NLogConfigurationException(message);
-                        }
-                        InternalLogger.Error(message);
-                    }
-                    if (!error)
-                        return;
+                TargetForm = Application.OpenForms[FormName];
+                if (TargetForm == null)
+                {
+                    InternalLogger.Info("Form {0} not found, waiting for ReInitializeAllTextboxes.", FormName);
+                    return;
                 }
-                else
+
+                TargetRichTextBox = FormHelper.FindControl<RichTextBox>(ControlName, TargetForm);
+                if (TargetRichTextBox == null)
                 {
-                    InternalLogger.Info("Form {0} not found, creating a new one", FormName);
+                    InternalLogger.Info("Rich text box control '{0}' cannot be found on form '{1}'. Waiting for ReInitializeAllTextboxes.", ControlName, FormName);
+                    return;
                 }
+
+                //actually attached to a target, all ok
+            }
+        }
+
+        /// <summary>
+        /// Called from constructor when error is detected. In case LogManager.ThrowExceptions is enabled, throws the exception, otherwise - logs the problem message
+        /// </summary>
+        /// <param name="message">exception/log text</param>
+        private static void HandleError(string message)
```

Comment:
IMO a handleError with message parameters would be nice (`params object[]`), and no string concat in the callers. DO you agree? (not very important)